src/ProductConstructionService/ProductConstructionService.WorkItems/LocalReplicaWorkItemProcessorStateWriterFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using ProductConstructionService.Common;
+
+namespace ProductConstructionService.WorkItems;
+public class LocalReplicaWorkItemProcessorStateWriterFactory : IReplicaWorkItemProcessorStateWriterFactory
+{
+    private readonly IRedisCacheFactory _redisCacheFactory;
+    private readonly ILogger<WorkItemProcessorStateWriter> _logger;
+
+    public LocalReplicaWorkItemProcessorStateWriterFactory(IRedisCacheFactory redisCacheFactory, ILogger<WorkItemProcessorStateWriter> logger)
```

Comment:
Were we saying we should use the primary constructors for new code?
(I don't remember)

Reply:
We did say that a while ago, but we diverged from there. It does fit in this class tho